### PR TITLE
🌱 Bump upgraded k8s version and add basic logging after e2e tests

### PR DESCRIPTION
**Pull request by lentzi90**
**What this PR does / why we need it**:

Due to changes in kubeadm and cluster-api for the container registry, we need to bump the upgraded k8s version to pass the test.

It is currently hard to tell what state the cluster was in when it failed. This should hopefully make it a bit clearer.
Similar to https://github.com/metal3-io/cluster-api-provider-metal3/pull/714 (but this was hard to backport it seems).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: test/e2e/common.go [hunk 1]
```diff
@@ -175,3 +177,99 @@ func deploymentRolledOut(ctx context.Context, clientSet *kubernetes.Clientset, n
...
+		Logf("%s	%s	%s	%s", machine.GetName(), machine.Status.GetTypedPhase(), providerID, *machine.Spec.Version)
+	}
+	Logf("---------------------------------------------------------------------------------")
+	Logf("%d Machines in total", len(machines.Items))
+	Logf("=================================================================================")
+}
+
+// listNodes logs the names, status and Kubernetes version of all Nodes.
+// Similar to kubectl get nodes.
+func listNodes(ctx context.Context, c client.Client) {
```
Comment (mquhuy) [inline test/e2e/common.go:255]: Any reasons this doesn't include opts?

Comment (lentzi90) [reply test/e2e/common.go:255]: Just no need for now. All the other resources we list are namespaced, so we need to at least say what namespace to look in. Nodes are cluster wide so no need. We could at some point add filters for control plane and workers but for now we don't use it for anything like that

**Review by kashifest — APPROVED** [review]

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**